### PR TITLE
`PollData` and `PollAnswer` rename

### LINKED ISSUE #536: Rename `PollData` to `Poll`
As the Clean architecture masters states like `Info` and `Data` are redundant if the original name can express the meaning of what it's describing. For this reason I suggest to rename `PollData` to `Poll`
cc. @dadamu , @RiccardoM

### PR DESCRIPTION
**Pull request by bragaz**
## Description
This PR renames `PollData` to `Poll` and `PollAnswer` to `Answer`.
Close #536

## Checklist
- [x] Targeted PR against correct branch.
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote unit tests.
- [ ] Wrote integration tests (simulation & CLI). 
- [x] Updated the documentation. 
- [x] Added an entry to the `CHANGELOG.md` file.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.

### CODE REVIEW

File: proto/desmos/posts/v1beta1/polls.proto [hunk 1]
```diff
@@ -6,14 +6,14 @@ import "google/protobuf/timestamp.proto";
 
 option go_package = "github.com/desmos-labs/desmos/x/staging/posts/types";
 
-// PollAnswer contains the data of a single poll answer inserted by the creator
-message PollAnswer {
+// Answer contains the data of a single poll answer inserted by the creator
+message Answer {
```
Comment (dadamu) [inline proto/desmos/posts/v1beta1/polls.proto:10]: @bragaz There is another object named `UserAnswer` to show an answer of a user for a poll. I think using `ProvidedAnswer` may be more explicit. What do you think?

Comment (RiccardoM) [reply proto/desmos/posts/v1beta1/polls.proto:10]: I agree with @dadamu that this should be more specific. Another way of achieving this is to move this message inside the `Poll` message:

```protobuf
message Poll {
  ...

  message Answer {
    ...
  }
}
```

Comment (bragaz) [reply proto/desmos/posts/v1beta1/polls.proto:10]: @dadamu @RiccardoM you're right, I will go with `ProvidedAnswer` then 👍

File: x/staging/posts/keeper/keeper_polls_test.go [hunk 2]
```diff
@@ -74,7 +74,7 @@ func (suite *KeeperTestSuite) TestKeeper_GetUserAnswersByPost() {
 			expStored: nil,
 		},
 		{
-			name: "Answers returned correctly",
+			name: "ProvidedAnswers returned correctly",
```
Comment (dadamu) [inline x/staging/posts/keeper/keeper_polls_test.go:77]:
```suggestion
			name: "UserAnswers returned correctly",
```

**Review by dadamu — APPROVED** [review]
It looks good to me.

**Review by RiccardoM — APPROVED** [review]

**Pull request merged**